File: 백준/Bronze/24510. 시간복잡도를 배운 도도/시간복잡도를 배운 도도.c
#include <stdio.h>
#include <string.h>

#define MAX_LEN 10002

int main() {
    int C;
    scanf("%d", &C);
    getchar();
    int dap = 0;

    for (int i = 0; i < C; i++) {
        char code[MAX_LEN];
        fgets(code, MAX_LEN, stdin);
        int tmp = 0;
        int j = 0;
        while (j < strlen(code)) {
            if (j + 3 <= strlen(code) && strncmp(&code[j], "for", 3) == 0) {
                tmp++;
                j+=3;
            }
            else if (j + 5 <= strlen(code) && strncmp(&code[j], "while", 5) == 0) {
                tmp++;
                j+=5;
            }
            else {
                j++;
            }
        }
        dap = (dap < tmp) ? tmp : dap;
    }
    printf("%d\n", dap);
    return 0;
}

File: 백준/Bronze/24510. 시간복잡도를 배운 도도/시간복잡도를 배운 도도.py
import sys
input = lambda: sys.stdin.readline().rstrip()


def main():
    C = int(input())
    dap = 0
    for _ in range(C):
        code = input()
        i = 0
        tmp = 0
        while i < len(code):
            if code[i:i+3] == 'for':
                tmp += 1
                i+=3
            elif code[i:i+5] == 'while':
                tmp += 1
                i+=5
            else:
                i+=1
        dap = tmp if dap < tmp else dap
    print(dap)


if __name__ == '__main__':
    main()